homeassistant/components/surepetcare/sensor.py

```diff
@@ -98,3 +102,37 @@ async def async_added_to_hass(self) -> None:
             async_dispatcher_connect(self.hass, TOPIC_UPDATE, self._async_update)
         )
         self._async_update()
+
+
+class SureLock(SensorEntity):
+    """A sensor implementation for Sure Petcare Entities."""
+
+    _attr_should_poll = False
+
+    def __init__(self, _id: int, spc: SurePetcareAPI) -> None:
+        """Initialize a Sure Petcare sensor."""
+        self._id = _id
+        self._spc: SurePetcareAPI = spc
+
+        surepy_entity: SurepyEntity = self._spc.states[_id]
+
+        self._attr_device_class = None
+        self._attr_name = f"{surepy_entity.type.name.capitalize()} {surepy_entity.name.capitalize()} Lock"
+        self._attr_unique_id = f"{surepy_entity.household_id}-{surepy_entity.id}-lock"
+
+    @callback
+    def _async_update(self) -> None:
+        """Get the latest data and update the state."""
+        surepy_entity = self._spc.states[self._id]
+        state = surepy_entity.state
+        self._attr_state = state
+
+        self.async_write_ha_state()
+        _LOGGER.debug("%s -> state: %s", self.name, state)
```

Comment:
Is this necessary? Won't a `STATE_CHANGED` event get fired (and a corresponding log written)?

Reply:
I just copied the code from the battery sensor. Happy to remove the state change request if not required. The logging is just a debug log as in the battery sensor.

Reply:
My comment was specifically for the logging statement.

Reply:
The state change won't add it to the log (as shown in Configuration/Logs), it will add it to the event stream as far as I am aware. I'm happy to remove it, as I said I just copied the battery sensor.

Reply:
All state changes are logged at debug level by the core.